### PR TITLE
[FIX] web_editor: only allow to embed images when composing a mail

### PR DESCRIPTION
**Pull request by len-foss**
Embedding a video can't work because it embeds an iframe, which is not supported
by most email clients for security reasons.
(What happens when Gmail shows embedded videos in displayed emails, is that you
send a link in the email body, and it's at rendering that Gmail adds the iframe
into the page.)
So the bugfix solution is to hide the button.

Since there is an options to only allow to add images, and an option "res_model"
that allows us to know when we are in that case, we remove the video option.

Essentially a foward-port of v11 commit.

opw 1905497

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web_editor/static/src/js/widgets/widgets.js [hunk 1]
```diff
@@ -1135,10 +1135,11 @@ var MediaDialog = Dialog.extend({
 
         this.multiImages = options.multiImages;
         var onlyImages = options.onlyImages || this.multiImages || (this.media && (this.$media.parent().data('oeField') === 'image' || this.$media.parent().data('oeType') === 'image'));
+        var mailCompose = options.res_model === "mail.compose.message";
         this.noImages = options.noImages;
-        this.noDocuments = onlyImages || options.noDocuments;
+        this.noDocuments = onlyImages || options.noDocuments || mailCompose;
```
Comment (nim-odoo) [inline addons/web_editor/static/src/js/widgets/widgets.js]: how is this related to videos?

**Comment by len-odoo** [comment]
@qsm-odoo Thank you for you review.
The v11 isn't merged yet, so the commit reference doesn't truly exist yet.
Also, in v12 the possibilities are more fine-grained, so you can create a media dialog without videos, but you can't in v11 (there's onlyImages or everything).

So this new version follows your advice (correctly I hope).

File: addons/web_editor/static/src/js/editor/rte.summernote.js [hunk 2]
```diff
@@ -1230,12 +1230,16 @@ var SummernoteManager = Class.extend(mixins.EventDispatcherMixin, {
...
+        var mailCompose = res_model === "mail.compose.message";
+        var noVideos = data.options.noVideos || mailCompose;
 
         var mediaDialog = new weWidgets.MediaDialog(this,
             _.extend({
-                res_model: data.$editable.data('oe-model'),
+                res_model: res_model,
                 res_id: data.$editable.data('oe-id'),
                 domain: data.$editable.data('oe-media-domain'),
+                noVideos: noVideos,
```
Comment (qsm-odoo) [inline addons/web_editor/static/src/js/editor/rte.summernote.js]: I think it is wrong since if `data.options.noVideos` is `false` then you will ignore `mailCompose` since the `extend` will overridde the default values here with `data.options`.

Anyways, here you are in the code which handles a dialog opening demand. It is not supposed to ignore the demands and change them. The code which really should have the control is the one making that demand, which is the `showImageDialog` function.

Comment (len-odoo) [reply addons/web_editor/static/src/js/editor/rte.summernote.js]: All right.

Comment (len-odoo) [reply addons/web_editor/static/src/js/editor/rte.summernote.js]: Updated with commit reference.

**Comment by len-odoo** [comment]
robodoo r+
